bin/check_google_doc_export.rb

```diff
@@ -16,12 +16,22 @@
 require_relative '../lib/article_json'
 
 doc_id = '1E4lncZE2jDkbE34eDyYQmXKA9O26BHUiwguz4S9qyE8'
-url = "https://docs.google.com/feeds/download/documents/export/Export?id=#{doc_id}&exportFormat=html"
+url =
+  "https://docs.google.com/feeds/download/documents/export/Export?id=#{doc_id}&exportFormat=html"
 expected_document = JSON.parse(File.read('spec/fixtures/reference_document_parsed.json'))
 
 exported_doc = Net::HTTP.get(URI.parse(url))
 document = JSON.parse(ArticleJSON::Article.from_google_doc_html(exported_doc).to_json)
```

Comment:
Not your change, but is there a reason for serializing and deserializing to & from JSON? Could that work too?
```suggestion
document = ArticleJSON::Article.from_google_doc_html(exported_doc).to_h
```

Reply:
I will try it

Reply:
So the 'good reason' is to compare like with like.

Reply:
Alternatively I could move the parsing into the `nullify` method but it is not really any better placed there IMO, at least not without the method being renamed>

```ruby
doc_export = ArticleJSON::Article.from_google_doc_html(exported_doc).to_json

expected_document = File.read('spec/fixtures/reference_document_parsed.json')

# `source_url` (for hosted images) is dynamic, so we need to remove it from the comparison`
def nullify_source_url(hash)
  hash = JSON.parse(hash)
  hash['content'].each { |element| element['source_url'] = nil if element['source_url'] }
  hash
end
```

Reply:
> So the 'good reason' is to compare like with like.

I don't get what you mean. Unless I'm missing something, we have a hash, convert it to json (`.to_json`), then convert it back into a hash again (`JSON.parse`).
My point was to ideally work with the hash directly. Splitting the steps up would obfuscate it even further, so I don't see a good reason for that...

Reply:
I guess one reason could be that the parsed JSON would give you string keys, while the internal hash has Symbol keys? But then working with `transform_keys` might be clearer?

Reply:
Would this work?
```ruby
exported_article = ArticleJSON::Article.from_google_doc_html(Net::HTTP.get(URI.parse(url)))
expected_article = ArticleJSON::Article.from_json(File.read('spec/fixtures/reference_document_parsed.json'))

exported_article == expected_article

Reply:
> I guess one reason could be that the parsed JSON would give you string keys, while the internal hash has Symbol keys?

That was what I meant with 'like' with 'like'. Sorry for not being more explicit

There is deep nesting and do not have Rails `hash_with_indifferent` nor `deep_transform_keys` available to us.